config/impl/src/main/java/org/jboss/windup/config/loader/WindupRuleLoaderImpl.java

```diff
@@ -151,9 +155,12 @@ private WindupRuleMetadata build(GraphContext context, Predicate<WindupRuleProvi
             for (final Rule rule : rules)
             {
                 i++;
+
+                // Metadata
                 if (rule instanceof Context)
-                    provider.enhanceMetadata((Context) rule);
+                    processRuleMetadata(rule, provider);
 
+                // Rule ID
```

Comment:
IMO these are not unnecessary. They fulfill the purpose of comments - to make it clear what's going on, visually demarcate important parts and describe the original intend of given part of code. You don't think so?